### ISSUE TITLE
Right-click on another message in the list acts on multiple messages

### ISSUE THREAD

**Issue by dapphp** [JohnDoh/Roundcube-Plugin-Context-Menu#77]
Experiencing issue in Roundcube 1.1.4 with Context Menu 2.1.2 and latest Git branch.

**Symptoms:**
Right clicking on a previously unselected message and performing an action on it (e.g. Mark as Read) acts on both the right-clicked on message, as well as the previously selected message.

**Expected Behavior:**
Only the message that was right-clicked on should be affected by the context menu action

**Actual Behavior:**
The message that was right-clicked on as well as the previously selected message row are acted on.

**Steps to Reproduce:**
To illustrate the example, disable the preview pane so messages aren't automatically marked read, and have a handful of messages either all read or all unread.

- Click/select the first message in the message list (state Unread)
- Right click on any other message in the list
- The first selected message appears to be unselected (message row is un-highlighted)
- Context menu appears on the other message that was right-clicked on
- From the context menu, choose Mark >> As Read

At this point, the message that was right clicked on is marked read *in addition to* the first message that was selected.  It is expected that the first message should not be affected.

This also happens if you try to move a message or forward.  Instead of moving or forwarding the single message, both messages are moved or forwarded.

If multiple messages had been previously selected and an unselected message is right-clicked on, all previously selected messages in addition to the right-clicked on message are affected.

**Fix:**
I've created this [Patch](https://gist.github.com/dapphp/9c94d1b9782397c22ecc3a285d079133) which is only partially tested at this point, that seems to fix the problem.

The patch works by checking to see if the `uid` of the context menu target is a selected message in the RC message list, if it is not, it calls select_row on the target message, clearing any previous selections.

If the message *is* in the list of selected rows, no special action is taken (this allows one to multiselect messages and right click on one of the selections to have the context menu act on all selections, or if only one row is selected and they right-click on the selected row, the context menu still only acts on the selected row).

**Comment by JohnDoh** [comment]
This is with the Larry skin, right?

I followed your steps but I do not get the same result, you said "The first selected message appears to be unselected (message row is un-highlighted)". This does not happen for me. This plugin does trigger some list events so may be you are running some other plugin and the two are not playing nicely together?

**Comment by dapphp** [comment]
It was with the larry skin I was experiencing it, using dev-release-2.1.2 (not dev-master).

We have since upgraded to Roundcube 1.1.5 and fixed an issue with the message list and I cannot replicate the issue now as well.  Also could not replicate with a fresh install of RC 1.1.5 and only the contextmenu plugin.

It must have been an interaction with another plugin as you said or an issue with list.js that has since been fixed in 1.1.5.

Sorry for the false report!
